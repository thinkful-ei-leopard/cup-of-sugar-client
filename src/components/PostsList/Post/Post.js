import React from 'react';
import styles from './Post.module.scss';
import cx from 'classnames';
import { Link } from 'react-router-dom';
import UserContext from '../../../contexts/UserContext'
import Button from '../../Button/Button'
import PostsApiService from '../../../services/posts-api-service'

export default class Post extends React.Component {

  static contextType = UserContext

  render() {
    const { post } = this.props;

    let title =
      post.title.length < 42 ? post.title : post.title.slice(0, 35) + '...';

    if(this.context.user.id === post.user_id) {
      return (
        <Link to={`/post/${post.id}`}>
          <li className={styles.Post}>
            {/* <div className={styles.typeCircle}></div> */}
            <span
              className={cx(
                styles.postTitle,
                styles.postEl,
                post.type === 'offer' ? styles.offerStyle : styles.requestStyle
              )}>
              {title}
            </span>
            <span className={cx(styles.postType, styles.postEl)}>
              {post.type}
            </span>
            <span className={cx(styles.postComments, styles.postEl)}>
              {post.comments}
            </span>
            <span className={cx(styles.postUserName, styles.postEl)}>
              {post.name}
            </span>
            <span className={cx(styles.postDateFull, styles.postEl)}>
              {post.date_modified.slice(0, 10)}
            </span>
            <span className={cx(styles.postDateTrim, styles.postEl)}>
              {post.date_modified.slice(5, 10)}
            </span>
            <Button onClick={(e) => {
              e.preventDefault()
              PostsApiService.deletePost(post.id)}
              }type='delete' title='Delete' className={styles.deletePostButton} id={styles.deletePostButton}>X</Button>
          </li>
        </Link>
      );
    }
    return (
      <Link to={`/post/${post.id}`}>
        <li className={styles.Post}>
          {/* <div className={styles.typeCircle}></div> */}

          <span
            className={cx(
              styles.postTitle,
              styles.postEl,
              post.type === 'offer' ? styles.offerStyle : styles.requestStyle
            )}>
            {title}
          </span>
          <span className={cx(styles.postType, styles.postEl)}>
            {post.type}
          </span>
          <span className={cx(styles.postComments, styles.postEl)}>
            {post.comments}
          </span>
          <span className={cx(styles.postUserName, styles.postEl)}>
            {post.name}
          </span>
          <span className={cx(styles.postDateFull, styles.postEl)}>
            {post.date_modified.slice(0, 10)}
          </span>
          <span className={cx(styles.postDateTrim, styles.postEl)}>
            {post.date_modified.slice(5, 10)}
          </span>
          <Button type='delete' className={styles.deletePostButton} id={styles.placeholderInvisibleButton}>X</Button>
        </li>
      </Link>
    );
    
  }
}
