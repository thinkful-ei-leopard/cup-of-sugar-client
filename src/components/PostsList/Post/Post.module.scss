@import '../../../styles/_variables';
@import '../../../styles/_breakpoints';

.Post {
  display: flex;
  justify-content: space-between;
  padding: 0.5rem 0;
  margin: 0 auto;
  transition: background-color 0.2s;
  border-radius: 5px;
  &:hover {
    background-color: var(--post-background);
  }
}

@include mobilePortrait {
  .Post {
    padding: 0.5rem 1rem;
  }
}

.postEl {
  margin-right: 1rem;
  text-align: center;
  display: inline-block;
}

.typeCircle {
  display: inline-block;
  width: 10px;
  height: 10px;
  margin: 0.5rem;
  border-radius: 50%;
}

.postTitle {
  width: 360px;
  min-width: 315px;
  text-align: left;
  display: flex;
}

.postType {
  min-width: 90px;
  display: none;
}

@include laptop {
  .postType {
    display: inline-block;
  }
}

.offerStyle,
.requestStyle {
  display: flex;
  justify-content: flex-start;
}

@media (max-width: 600px) {
  .postTitle {
    min-width: 150px;
  }
}

.offerStyle {
  color: var(--dark1);
}

.requestStyle {
  color: var(--dark2);
}

.postComments {
  width: 170px;
  display: none;
}

@include tabletLandscape {
  .postComments {
    display: inline-block;
  }
}

.postUserName {
  min-width: 140px;
  display: none;
}

@media (min-width: 800px) {
  .postUserName {
    display: inline-block;
  }
}

.postDateFull {
  width: 160px;
  display: none;
}

.postDateTrim {
  width: 160px;
  min-width: 52px;
  display: inline-block;
  margin-right: 0;
}

@include tabletLandscape {
  .postDateFull {
    display: inline-block;
  }
  .postDateTrim {
    display: none;
  }
}

#deletePostButton {
  padding-right: 0.5rem;
  padding-top: 0.25rem;
  padding-left: 0.5rem;
  padding-bottom: 0.25rem;
  color: var(--black);
  margin-left: 0.1rem;
  background-color: var(--post-background);
  border: none;
  font-weight: bold;
  font-size: 0.7rem;
  &:hover {
    color: var(--warning);
  }
}

#placeholderInvisibleButton {
  padding-right: 0.5rem;
  padding-top: 0.25rem;
  padding-left: 0.5rem;
  padding-bottom: 0.25rem;
  border: none;
  color: white;
  background-color: white;
  opacity: 0;
}

.resolvedStyle {
  text-decoration: line-through;
}
