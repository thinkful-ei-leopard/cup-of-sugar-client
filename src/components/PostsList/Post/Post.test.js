import React from 'react';
import ReactDOM from 'react-dom';
import { MemoryRouter as Router } from 'react-router-dom';
import Post from './Post';

it('renders without crashing', () => {
  const div = document.createElement('div');

  ReactDOM.render(
    <Router>
      <Post
        className={undefined}
        deletePost={() => {}}
        post={{ title: '', date_modified: '2020-05-04T19:25:55.864Z' }}
      />
    </Router>,
    div
  );

  ReactDOM.unmountComponentAtNode(div);
});
