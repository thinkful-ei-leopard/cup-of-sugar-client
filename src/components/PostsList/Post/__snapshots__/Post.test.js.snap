// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<li
  className="Post"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <span
    className="postTitle postEl"
  >
    <a
      className="requestStyle"
      href="/post/undefined"
      onClick={[Function]}
    >
      
    </a>
  </span>
  <span
    className="postType postEl"
  />
  <span
    className="postComments postEl"
  />
  <span
    className="postUserName postEl"
  />
  <span
    className="postDateFull postEl"
  >
    2020-05-04
  </span>
  <span
    className="postDateTrim postEl"
  >
    05-04
  </span>
</li>
`;
